pulp_rpm/tests/functional/api/test_rbac_crud.py

```diff
@@ -0,0 +1,317 @@
+        monitor_task(sync_res.task)
+        repo = rpm_repository_api.read(repo.pulp_href)
+        assert repo.latest_version_href.endswith("/1/")
+
+    # Remove
+    with user_two, pytest.raises(ApiException) as exc:
+        rpm_repository_api.delete(remote.pulp_href)
+
+    assert exc.value.status == 404
+
+    with user_one:
+        response = rpm_rpmremote_api.delete(remote.pulp_href)
+        monitor_task(response.task)
+        response = rpm_repository_api.delete(repo.pulp_href)
+        monitor_task(response.task)
+        assert rpm_rpmremote_api.list().count == 0
+        assert rpm_repository_api.list().count == 0
+
+
+def test_rbac_acs(gen_user, rpm_acs_api, rpm_rpmremote_api):
+    """Test RPM ACS CRUD."""
+    user_one = gen_user(
+        model_roles=[
+            "rpm.rpmalternatecontentsource_creator",
+            "rpm.rpmremote_owner",
+        ]
+    )
+    user_two = gen_user(
+        model_roles=[
+            "rpm.rpmremote_owner",
+        ]
+    )
+
+    acs = None
+    remote_data = gen_rpm_remote(policy="on_demand")
+    remote = rpm_rpmremote_api.create(remote_data)
+
+    acs_data = {
+        "name": uuid4(),
+        "remote": remote.pulp_href,
+    }
+
+    # Create
+    with user_one:
+        acs = rpm_acs_api.create(acs_data)
+        assert rpm_acs_api.list().count == 1
+
+    with user_two, pytest.raises(ApiException):
+        rpm_acs_api.create(acs_data)
+
+    # Update & Read
+    # Presume that if you can't read you can't do any other operations
+    with user_two, pytest.raises(ApiException):
+        rpm_acs_api.read(acs.pulp_href)
+    with user_one:
+        acs_to_update = rpm_acs_api.read(acs.pulp_href)
+        acs_to_update.paths[0] = "files/"
+        response = rpm_acs_api.update(acs_to_update.pulp_href, acs_to_update)
+        monitor_task(response.task)
+        assert rpm_acs_api.list().count == 1
+        assert "files/" in rpm_acs_api.read(acs.pulp_href).paths
+
+    # Remove
+    with user_two, pytest.raises(ApiException):
+        rpm_acs_api.delete(acs.pulp_href)
+    with user_one:
+        rpm_acs_api.delete(acs.pulp_href)
+        rpm_rpmremote_api.delete(remote.pulp_href)
+        assert rpm_acs_api.list().count == 0
+
+
+def test_rbac_publication(gen_user, rpm_rpmremote_api, rpm_repository_api, rpm_publication_api):
+    """Test RPM publication CRUD."""
+    user_one = gen_user(
+        model_roles=[
+            "rpm.rpmpublication_creator",
+            "rpm.rpmremote_owner",
+            "rpm.rpmrepository_owner",
+        ]
+    )
+    user_two = gen_user(
+        model_roles=[
+            "rpm.rpmremote_owner",
+            "rpm.rpmrepository_owner",
+        ]
+    )
+
+    publication = None
+    remote_data = gen_rpm_remote(RPM_UNSIGNED_FIXTURE_URL)
+    remote = rpm_rpmremote_api.create(remote_data)
+    repo = rpm_repository_api.create(gen_repo())
+    sync_url = RpmRepositorySyncURL(remote=remote.pulp_href)
```

Comment:
i think, if you want to spare on couple of api calls, like remote creation and repo sync, one can assign repo version 0 to publication